### PR TITLE
Updating the installation console output

### PR DESCRIPTION
**Pull request by joecheuk**

### CODE REVIEW

File: setup/gcp/installer/forseti_installer.py [hunk 1]
```diff
@@ -44,13 +44,31 @@ def __init__(self):
...
 
-    def run_setup(self):
-        """Run the setup steps."""
+    def run_setup(self,
+                  setup_continuation=False,
+                  last_installation=True,
+                  previous_instructions=None):
+        """Run the setup steps.
+
+        if setup_continuation is True, we don't need to run the pre-flight
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: nit: If

File: setup/gcp/installer/forseti_installer.py [hunk 2]
```diff
@@ -44,13 +44,31 @@ def __init__(self):
...
+        Returns:
+            list: List of all instructions.
+        """
         utils.print_banner('Installing Forseti {} v{}'.format(
             self.config.installation_type, utils.get_forseti_version()))
 
-        # Preflight checks.
-        self.preflight_checks()
+        if not setup_continuation:
+            # Fresh installation, run pre-flight checks.
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: This should be self documenting.

File: setup/gcp/installer/forseti_installer.py [hunk 3]
```diff
@@ -66,10 +84,21 @@ def run_setup(self):
...
-                                       bucket_name)
+        instructions = self.post_install_instructions(deploy_success,
+                                                      deployment_name,
+                                                      conf_file_path,
+                                                      bucket_name)
+
+        if previous_instructions is not None:
+            instructions = previous_instructions + instructions
+
+        if last_installation:
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: nit: last_installation isn't really descriptive. There's only one install?

File: setup/gcp/installer/forseti_installer.py [hunk 4]
```diff
@@ -66,10 +84,21 @@ def run_setup(self):
...
+            # Only print the instructions if this is the final installation
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: Comment shouldn't be needed.

File: setup/gcp/installer/forseti_installer.py [hunk 5]
```diff
@@ -152,14 +178,14 @@ def deploy(self, deployment_tpl_path, conf_file_path, bucket_name):
             if self.config.dry_run:
                 print('This is a dry run, will not copy any files.')
 
-            utils.print_banner('Copying configuration file/deployment'
-                               ' template to GCS')
+            utils.print_banner('Backing up important files to GCS')
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: nit: CamelCase, fix throughout for banners.

File: setup/gcp/installer/forseti_installer.py [hunk 6]
```diff
@@ -168,8 +194,9 @@ def deploy(self, deployment_tpl_path, conf_file_path, bucket_name):
             deployment_tpl_output_path = (
                 constants.DEPLOYMENT_TEMPLATE_OUTPUT_PATH.format(bucket_name))
 
-            print('Copying {} to {}'.format(deployment_tpl_path,
-                                            deployment_tpl_output_path))
+            print('Copying the Forseti {} deployment template to:\n    {}'
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: You can use the `\t` character to actually tab.

File: setup/gcp/installer/forseti_installer.py [hunk 7]
```diff
@@ -185,23 +212,24 @@ def wait_until_vm_initialized(self, vm_name):
...
-        utils.print_banner('{} VM Initialization'.format(
+        utils.print_banner('Forseti {} VM Initialization'.format(
             installation_type.capitalize()))
         _, zone, name = gcloud.get_vm_instance_info(vm_name)
 
         status_checker = (lambda: gcloud.check_vm_init_status(name, zone))
 
-        loading_message = ('This may take a few minutes. Waiting for Forseti '
-                           '{} to be initialized..'.format(installation_type))
+        loading_message = ('Waiting for Forseti {} to be initialized..'.format(
```
Comment (mirons-google) [inline setup/gcp/installer/forseti_installer.py]: nit: Just be consistent `... `, e.g. three dots and a space on loading messages. Fix throughout.

File: setup/gcp/installer/forseti_installer.py [hunk 8]
```diff
@@ -44,13 +44,31 @@ def __init__(self):
...
-            self.config.installation_type, utils.get_forseti_version()))
+    def run_setup(self,
+                  setup_continuation=False,
+                  last_installation=True,
+                  previous_instructions=None):
+        """Run the setup steps.
 
-        # Preflight checks.
-        self.preflight_checks()
+        if setup_continuation is True, we don't need to run the pre-flight
```
Comment (blueandgold) [inline setup/gcp/installer/forseti_installer.py]: First word of a sentence should be capitalize: `If`

File: setup/gcp/installer/forseti_installer.py (hunk 3 above)
Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: Updated to final_setup

File: setup/gcp/installer/forseti_installer.py (hunk 4 above)
Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: Comment removed.

File: setup/gcp/installer/forseti_installer.py (hunk 7 above)
Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: Updated to have 3 dots

File: setup/gcp/installer/forseti_installer.py (hunk 6 above)
Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: Updated to use \t (8 spaces on gcloud console.)

File: setup/gcp/installer/forseti_installer.py (hunk 5 above)
Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: Updated.

File: setup/gcp/installer/forseti_installer.py (hunk 1 above)
Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: Updated.

**Comment by joecheuk** [comment]
Resolves: #1188

File: setup/gcp/installer/forseti_installer.py [hunk 9]
```diff
@@ -26,6 +26,49 @@
...
+        """Merge instructions, input instructions will be merged to the head
+        of the current instructions.
+
+        Args:
+            other_instruction (ForsetiInstructions): The other instructions.
+        """
+        self.deployed_branches = (other_instruction.deployed_branches +
+                                  self.deployed_branches)
+        self.deployment_template_messages = (
+                other_instruction.deployment_template_messages +
```
Comment (blueandgold) [inline setup/gcp/installer/forseti_installer.py]: hanging indents are 4 spaces

File: setup/gcp/installer/forseti_installer.py [hunk 10]
```diff
@@ -26,6 +26,49 @@
 from util import utils
 
 
+class ForsetiInstructions(object):
+    """Forseti post-setup instructions."""
```
Comment (blueandgold) [inline setup/gcp/installer/forseti_installer.py]: this can handle more than just `post-setup`, right?

Comment (joecheuk) [reply setup/gcp/installer/forseti_installer.py]: I think so, will phrase it to setup instead of post-setup.

**Pull request merged**